src/vm/excep.cpp

```diff
@@ -243,7 +243,7 @@ ULONG GetExceptionMessage(OBJECTREF throwable,
 
 //-----------------------------------------------------------------------------
 // Given an object, get the "message" from it.  If the object is an Exception
-//  call Exception.InternalToString, otherwise, call Object.ToString
+//  call Exception.ToString, otherwise, call Object.ToString
```

Comment:
Can't we just always call Object.ToString() now?  I'm assuming it's virtual dispatch.

Reply:
oh ,yes of course

Reply:
Hmm, but it asserts that it is an exception: `  _ASSERTE(IsException(throwable->GetMethodTable()));   ` (and the method is named `GetExceptionMessage`)

I think I'll leave this cleanup for now.

Reply:
IL allows you to throw object of any type. It does not have to inherit from exception. This feature is used by e.g. managed C++. For languages that expect exceptions to inherit from Exception (like C#), the runtime wraps it in `RuntimeWrappedException` on demand. In general runtime code, it is best to assume that the exception is arbitrary object. I do not see anything that guarantees the wrapping to happen on this codepath.

Reply:
#24797